catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/fragment/DataListFragment.java

```diff
@@ -350,6 +361,209 @@ public void onItemLongClick(final UserData item, CheckableVH holder) {
+		List<UserList> globalLists = project.getUserLists();
+		Scene currentScene = projectManager.getCurrentlyEditedScene();
+
+		List<UserData> itemsToDelete = filterDeletableUserDataItems(selectedItems, globalVariables, globalLists, currentScene);
+
+		if (itemsToDelete.size() > 0) {
+			deleteItems(itemsToDelete);
+		}
+	}
+
+	private List<UserData> filterDeletableUserDataItems(List<UserData> selectedItems,
+			List<UserVariable> globalVariables, List<UserList> globalLists, Scene currentScene) {
+		List<UserData> itemsToDelete = new ArrayList<>();
+
+		for (UserData item : selectedItems) {
+			if (globalVariables.contains(item) || globalLists.contains(item)) {
+				if (!checkIfUserDataItemInUse(item, null) && !checkIfUserDataItemInDropDown(item, null)) {
+					itemsToDelete.add(item);
+				}
+			} else {
+				if (!checkIfUserDataItemInUse(item, currentScene) && !checkIfUserDataItemInDropDown(item, currentScene)) {
+					itemsToDelete.add(item);
+				}
+			}
+		}
+
+		return itemsToDelete;
+	}
+
+	private boolean checkIfUserDataItemInUse(UserData item, @Nullable Scene currentScene) {
+		List<Scene> sceneList;
+		Project project = projectManager.getCurrentProject();
+		Sprite currentSprite = projectManager.getCurrentSprite();
+
+		if (currentScene == null) {
+			sceneList = project.getSceneList();
+		} else {
+			sceneList = new ArrayList<>();
+			sceneList.add(currentScene);
+		}
+
+		for (Scene scene : sceneList) {
+			for (Sprite sprite : scene.getSpriteList()) {
+				if (sprite == currentSprite || currentScene == null) {
+					for (Script script : sprite.getScriptList()) {
+						List<Brick> flatList = new ArrayList();
+						script.addToFlatList(flatList);
+						for (Brick brick : flatList) {
+							if (brick instanceof FormulaBrick) {
+								FormulaBrick formulaBrick = (FormulaBrick) brick;
+								for (Formula formula : formulaBrick.getFormulas()) {
+									FormulaElement formulaTree = formula.getFormulaTree();
+									List<String> lists = new ArrayList<>();
+									List<String> variables = new ArrayList<>();
+									formulaTree.getVariableAndListNames(variables, lists);
+
+									if (item instanceof UserVariable && variables.contains(item.getName())) {
+										showInUseAlert(item);
+										return true;
+									} else if (item instanceof UserList && lists.contains(item.getName())) {
+										showInUseAlert(item);
+										return true;
+									}
+								}
+							}
+						}
+					}
+				}
+			}
+		}
+		return false;
+	}
+
+	private boolean checkIfUserDataItemInDropDown(UserData item, @Nullable Scene currentScene) {
+		List<Scene> sceneList;
+		Project project = projectManager.getCurrentProject();
+		Sprite currentSprite = projectManager.getCurrentSprite();
+
+		if (currentScene == null) {
+			sceneList = project.getSceneList();
+		} else {
+			sceneList = new ArrayList<>();
+			sceneList.add(currentScene);
+		}
+
+		for (Scene scene : sceneList) {
```

Comment:
This seems to be duplicate code. Please try to define some logic you could use in both of your cases, or try to figure out an existing function like discussed via call.